service/src/test/java/greencity/service/HabitAssignServiceImplTest.java

```diff
@@ -626,4 +629,79 @@ void updateShoppingItem() {
         assertEquals(ShoppingListItemStatus.ACTIVE, userShoppingListItem.getStatus());
 
     }
+
+    @Test
+    void saveUserShoppingListItems() {
+        List<ShoppingListItem> shoppingList = new ArrayList<>();
+        try {
+            Method method = habitAssignService
+                .getClass().getDeclaredMethod("saveUserShoppingListItems", List.class, HabitAssign.class);
+            method.setAccessible(true);
```

Comment:
Why do you use Reflection API? It's not recommended to do in the tests

Reply:
I haven't seen any PowerMock dependency in pom.xml, so I decided to use Reflection here.
In which way, I should fix it?

Reply:
Just add PowerMock dependency

Reply:
It is bad practice to use reflection for testing private methods. Private methods should be tested using public methods.
Method `saveUserShoppingListItems` can be tested using `assignCustomHabitForUser` and `assignDefaultHabitForUser` methods where it is  used